lib/chef/dsl/recipe.rb

```diff
@@ -118,6 +119,11 @@ def build_resource(type, name, created_at=nil, &resource_attrs_block)
         # Evaluate resource attribute DSL
         resource.instance_eval(&resource_attrs_block) if block_given?
 
+        # emit a cloned resource warning if it is warranted
+        if !trivial_prior_resource.nil? && ( !trivial_prior_resource || !resource.identicalish_resource?(cloned_resource))
```

Comment:
don't try to understand it right now, its a mess --  but this isn't the prior_resource its the return value of identicalish_resource?(prior_resource) check deep inside the load_prior_resource call.

i sorta want to bust this out into its own class (ResourceBuilder or something) and extract all the code from the DSL and all the code from the resource, but the `@params` ivar will probably troll me.